### PR TITLE
Added Documentation for QMC5883L + HMC5883L Doc improvements

### PR DESCRIPTION
**Pull request by timpur**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

### CODE REVIEW

File: components/sensor/hmc5883l.rst [hunk 1]
```diff
@@ -43,10 +46,63 @@ Configuration variables:
...
+- **range** (*Optional*): The range parameter for the sensor.
 - **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
-
 - **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
 
+Oversampling Options
+--------------------
+
+By default, the HMC5883L sensor measures each value 1 times when requesting a new value. You can, however,
+configure this amount. Possible sampling values:
```
Comment (OttoWinter) [inline components/sensor/hmc5883l.rst]: Add how this affects the values here (i.e. this averages x values together).

File: components/sensor/hmc5883l.rst [hunk 2]
```diff
@@ -43,10 +46,63 @@ Configuration variables:
...
+-  ``1x`` (default)
+-  ``2x``
+-  ``4x``
+-  ``8x``
+
+Data Rate Options
+-----------------
+
+By default, the HMC5883L sensor measures 15 sampled ever second. You can, however,
+configure this amount. Possible sampling values:
```
Comment (OttoWinter) [inline components/sensor/hmc5883l.rst]:
```suggestion
configure this amount. Possible data rate values:
```

We should not expose this directly to the user, we can calculate the best setting here ourselves by looking at the update_interval value.

File: components/sensor/hmc5883l.rst [hunk 3]
```diff
@@ -43,10 +46,63 @@ Configuration variables:
...
+-  ``7.5Hz``
+-  ``15Hz`` (default)
+-  ``30Hz``
+-  ``75Hz``
+
+Measurement Mode Options
+------------------------
+
+By default, the HMC5883L sensor measurement mode is normal. You can, however,
+configure this amount. Possible sampling values:
```
Comment (OttoWinter) [inline components/sensor/hmc5883l.rst]: Please explain what measurement mode does, and why the user would need it.

The philosophy here is: don't expose all settings the sensor has, only what users may want to change.

File: components/sensor/hmc5883l.rst [hunk 4]
```diff
@@ -43,10 +46,63 @@ Configuration variables:
...
+-  ``neg_bias``
+
+Range Options
+-------------
+
+By default, the HMC5883L sensor measurement range is 130uT. You can, however,
+configure this amount. Possible sampling values:
+
+-  ``88uT``
+-  ``130uT`` (default)`
```
Comment (OttoWinter) [inline components/sensor/hmc5883l.rst]:
```suggestion
-  ``130uT`` (default)
```

File: components/sensor/hmc5883l.rst [hunk 5]
```diff
@@ -43,10 +46,63 @@ Configuration variables:
...
+
+-  ``normal`` (default)
+-  ``pos_bias``
+-  ``neg_bias``
+
+Range Options
+-------------
+
+By default, the HMC5883L sensor measurement range is 130uT. You can, however,
+configure this amount. Possible sampling values:
```
Comment (OttoWinter) [inline components/sensor/hmc5883l.rst]:
```suggestion
configure this amount. Possible values:
```

**Comment by timpur** [comment]
Thanks @OttoWinter, will try get this done this weekend.

**Comment by timpur** [comment]
@OttoWinter Believe this is ready fir review again thx :)

File: components/sensor/qmc5883l.rst [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+.. seo::
+    :description: Instructions for setting up QMC5883L IMU compass sensors.
+    :image: qmc5883l.jpg
+    :keywords: QMC5883L
+
+The ``qmc5883l`` allows you to use your QMC5883L triple-axis magnetometers
+(`datasheet <http://wiki.sunfounder.cc/images/7/72/QMC5883L-Datasheet-1.0.pdf>`__) with
+ESPHome. The :ref:`I²C Bus <i2c>` is
+required to be set up in your configuration for this sensor to work.
```
Comment (OttoWinter) [inline components/sensor/qmc5883l.rst]: Please add a short paragraph below this one explaining the difference between this sensor and the HMC5883L.

**Review by OttoWinter — APPROVED** [review]

File: components/sensor/qmc5883l.rst [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  :ref:`Sensor <config-sensor>`.
+- **range** (*Optional*): The range parameter for the sensor.
+- **oversampling** (*Optional*): The oversampling parameter for the sensor.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+
+Range Options
+-------------
+
+By default, the QMC5883L sensor measurement range is 130uT. You can, however,
```
Comment (pilotak) [inline components/sensor/qmc5883l.rst]: typo - default is 200 not 130

Comment (timpur) [reply components/sensor/qmc5883l.rst]: thx

**Pull request merged**